pkg/agent/utils.go

```diff
@@ -465,6 +465,13 @@ func getAKSKubeletConfiguration(kc map[string]string) *datamodel.AKSKubeletConfi
 		ResolverConfig:                 kc["--resolv-conf"],
 		ContainerLogMaxSize:            kc["--container-log-max-size"],
 	}
+
+	// Serialize Image Pulls will only be set for k8s >= 1.31, currently RP doesnt pass this flag
+	// It will starting with k8s 1.31
+	if value, exists := kc["--serialize-image-pulls"]; exists {
+		kubeletConfig.SerializeImagePulls = strToBoolPtr(value)
```

Comment:
did we also wanna add a UT for this?

Reply:
with the original change I had, this was already UTed, but since I reverted it, added UTs specifically for this now